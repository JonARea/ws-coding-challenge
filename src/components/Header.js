import React from 'react'
import {styleHeader} from '../styles'
import {Link} from 'react-router-dom'

const Header = ({location}) => (
  <div style={styleHeader}>
    <h1 className="headerBrand">
      west elm
    </h1>
      {location.pathname !== '/' &&
        <Link to={'/'} style={{textDecoration: 'none', color: 'black'}}>
          <h4>All new Products</h4>
        </Link>
      }
  </div>
)

export default Header
