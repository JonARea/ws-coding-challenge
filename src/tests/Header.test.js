import React from 'react'
import Header from '../components/Header'
import renderer from 'react-test-renderer'
import {BrowserRouter} from 'react-router-dom'

test('Header shows Link to All products when on a single product page', () => {
  const noLink = renderer.create(
    <BrowserRouter>
      <Header location={{pathname: '/'}} />
    </BrowserRouter>
  )
  let tree = noLink.toJSON()
  expect(tree).toMatchSnapshot()

  const link = renderer.create(
    <BrowserRouter>
      <Header location={{pathname: '/somethingelse'}} />
    </BrowserRouter>
  )
  tree = link.toJSON()
  expect(tree).toMatchSnapshot()

})
