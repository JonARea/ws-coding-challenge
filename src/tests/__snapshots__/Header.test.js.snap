// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header shows Link to All products when on a single product page 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "borderBottom": "solid black 2px",
      "display": "flex",
      "flexDirection": "column",
      "width": "100vw",
    }
  }
>
  <h1
    className="headerBrand"
  >
    west elm
  </h1>
</div>
`;

exports[`Header shows Link to All products when on a single product page 2`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "borderBottom": "solid black 2px",
      "display": "flex",
      "flexDirection": "column",
      "width": "100vw",
    }
  }
>
  <h1
    className="headerBrand"
  >
    west elm
  </h1>
  <a
    href="/"
    onClick={[Function]}
    style={
      Object {
        "color": "black",
        "textDecoration": "none",
      }
    }
  >
    <h4>
      All new Products
    </h4>
  </a>
</div>
`;
